picard/script/functions.py

```diff
@@ -1208,10 +1210,11 @@ def func_map(parser, multi, loop_code, separator=MULTI_VALUED_JOINER):
     for loop_count, value in enumerate(multi_value, 1):
         func_set(parser, '_loop_count', str(loop_count))
         func_set(parser, '_loop_value', str(value))
+        # Make changes in-place
         multi_value[loop_count - 1] = str(loop_code.eval(parser))
     func_unset(parser, '_loop_count')
     func_unset(parser, '_loop_value')
-    return str(multi_value)
+    return multi_value.separator.join([x for x in multi_value if x])
```

Comment:
Instead, we could modify `MultiValue.__str__()` to skip empty values (if those aren't expected anyway). It would reduce code redundancy.

```python
    def __str__(self):
        return self.separator.join([x for x in self if x])
```

https://github.com/metabrainz/picard/blob/ba862a14f8ec2df8f363e399f4c0bd64bdcf0bb5/picard/script/parser.py#L400


Not sure if there's any drawbacks. Thoughts?

Reply:
That actually makes a lot of sense to me, assuming that you always want to strip out empty elements (which it sounds like we do).  If we decide to go that way, should I make the change in this PR or create a separate one?

Reply:
I would do it in this PR, but let's wait feedback from @phw about this

Reply:
I agree that this is a good idea. I can't see any drawbacks currently